src/transformers/pipelines/image_classification.py

```diff
@@ -184,8 +184,9 @@ def postprocess(self, model_outputs, function_to_apply=None, top_k=5):
             top_k = self.model.config.num_labels
 
         outputs = model_outputs["logits"][0]
-        if self.framework == "pt" and outputs.dtype in (torch.bfloat16, torch.float16):
-            outputs = outputs.to(torch.float32).numpy()
+        if self.framework == "pt" and outputs.dtype == torch.bfloat16:
+            # To enable using bf16
+            outputs = outputs.float().numpy()
```

Comment:
This seems like a regression to me! The previous code worked with both `torch.bfloat16` and `torch.float16`.

Reply:
I use torch 2.4 in python 3.10 and it failed:
![image](https://github.com/user-attachments/assets/e5bf0154-b975-45fd-9608-0a2c4ab85715)